elkodon/src/lib.rs

```diff
@@ -88,7 +88,8 @@
 //!
 //! while !SignalHandler::termination_requested() {
 //!     let mut sample = publisher.loan()?;
-//!     unsafe { sample.as_mut_ptr().write(1234) };
+//!     sample.payload_mut().write(1234);
+//!     let sample = unsafe { sample.assume_init() };
```

Comment:
We still require `unsafe` in this case. Couldn't we maybe return an initialized sample in `write` so that we have something like
```rust
let initialized_sample = sample.payload_mut().write(1234);
```

Reply:
I also thought about this but the API would look a bit different since `payload_mut`returns a `MaybeUninit<T>` and `write`has this signature ` fn write(&mut self, val: T) -> &mut T`.

What we could do is to introduce `fn write_payload(self, val: MessageType) -> SampleMut<..., Header, MessageType>` in `SampleMut`. Could also be just `write` but `write_payload` is a bit less ambiguous. I'm don't have a strong opinion on the name though.